records_mover/session.py

```diff
@@ -190,9 +220,58 @@ def _boto3_session(self) -> Optional['boto3.session.Session']:
             return None
 
         if self._default_aws_creds_name is None:
-            return boto3.session.Session()
+            self.__boto3_session = boto3.session.Session()
         else:
-            return self.creds.boto3_session(self._default_aws_creds_name)
+            self.__boto3_session = self.creds.boto3_session(self._default_aws_creds_name)
+        return self.__boto3_session
+
+    def _gcs_creds(self) -> Optional['google.auth.credentials.Credentials']:
+        if self.__gcs_creds is not NotYetFetched.token:
+            return self.__gcs_creds
+
+        try:
+            import google.auth.exceptions
+            if self._default_gcp_creds_name is None:
+                import google.auth
+                credentials, project = google.auth.default()
+                self.__gcs_creds = credentials
+            else:
+                creds = self.creds.gcs(self._default_gcp_creds_name)
+                self.__gcs_creds = creds
+                return creds
+        except (OSError, google.auth.exceptions.DefaultCredentialsError):
+            # Examples:
+            #   OSError: Project was not passed and could not be determined from the environment.
+            #   google.auth.exceptions.DefaultCredentialsError: Could not automatically determine
+            #     credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create
+            #     credentials and re-run the application. For more information, please see
+            #     https://cloud.google.com/docs/authentication/getting-started
+            logger.debug("google.cloud.storage not configured",
+                         exc_info=True)
+            self.__gcs_creds = None
+        return self.__gcs_creds
+
+    def _gcs_client(self) -> Optional['google.cloud.storage.Client']:
+        # TODO: cache
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker